src/query/crypt_opfunc.c

```diff
@@ -178,14 +162,7 @@ str_to_hex (THREAD_ENTRY * thread_p, const char *src, int src_len, char **dest_p
       return NULL;
     }
 
-  while (i < src_len)
-    {
-      item_num = (unsigned char) src[i];
-      dest[2 * i] = hextable[item_num / 16];
-      dest[2 * i + 1] = hextable[item_num % 16];
-      i++;
-    }
-
+  str_to_hex_prealloced (src, src_len, dest, dest_len, lettercase);
   *dest_p = dest;
   *dest_len_p = dest_len;
```

Comment:
Isn't it
`*dest_len_p = dest_len - 1;`
Our QA team reported that the case result related to sha2 during testcase execution appeared to contain additional characters. I think +1 is for allocating space for End-of-String. So, I think you seem to have to return the length of dest length minus 1 (to exclude length for EOS).
If you run
`csql -u dba -c "select sha2('abc', 224)" demodb`
on 11.0.0.0125 and 10.1.4.7800, you can find differences. csql result of 11.0.0.0125 **do not have sigle quote at the end** of the result.